File: js-pr/56/javascript/code-quality-error-handling/regex_matcher_36.js
// {fact rule=code-quality-error-handling@v1.0 defects=0}
function matchPattern(text, pattern) {
    try {
        const regex = new RegExp(pattern);
        return regex.exec(text);
    } catch (error) {
        console.error('Regex error:', error.message);
        return null;
    }
}
// {/fact}

File: python-pr/52/python/code-quality-error-handling/regex_matcher_36.py
# {fact rule=code-quality-error-handling@v1.0 defects=0}
def match_pattern(text, pattern):
    import re
    try:
        return re.match(pattern, text)
    except re.error as e:
        print(f"Regex error: {e}")
        return None
# {/fact}